src/node/settings.ts

```diff
@@ -0,0 +1,63 @@
+const fsPromises = require('fs/promises');
+const fs = require('fs');
+const path = require('path');
+
+import { defaultSettings, getSettingsDirectory } from './constants';
+import type { Settings } from './types';
+
+async function setSettings(settings: Settings) {
+  let newSettings: Settings = settings;
+  if (settings.apiKey) {
+    newSettings = {
+      ...settings,
+      apiKey: {
+        key: settings.apiKey.key,
+        encrypted: settings.apiKey.encrypted,
+      },
+    };
+  }
+
+  try {
+    await saveSettings(newSettings);
+  } catch (error) {
+    console.log(error);
+  }
+}
```

Comment:
I don't understand what's going on here? what's this gymnastics with `newSettings` and why?

Reply:
yes, not needed